packages/vue-pdf-viewer/src/components/PDFViewer.vue

```diff
@@ -147,6 +161,23 @@ export default {
                 this.customScale.label = (this.currentScale * 100).toFixed(0) + '%'
                 this.customScale.value = this.currentScale
             }
+        },
+        async downloadDocument() {
+            if (!this.allowDocumentDownload) {
+                return
+            }
+            const index = this.documentName.lastIndexOf(".pdf");
+            const editedDocumentName = this.documentName.substring(0, index) + "_PREVIEW" + this.documentName.substring(index);
+            if (this.source instanceof Uint8Array) {
+                this.pdfViewer.downloadManager.downloadData(this.source, editedDocumentName)
+            } else {
+                if (/^blob:/.test(this.source)) {
+                    let blob = await fetch(this.source).then(r => r.blob());
```

Comment:
This should probably be `const blob` instead of `let`. Also there is a mix of await and then still using the `then` callback method. Is there a way to just use await?

Reply:
It's cleaned up now, forgot to do it for this last bit